internal/parser/planparserv2/utils.go

```diff
@@ -787,3 +788,77 @@ func decodeUnicode(input string) string {
 		return string(rune(code))
 	})
 }
+
+func parseISODuration(durationStr string) (*planpb.Interval, error) {
+	iso8601DurationRegex := regexp.MustCompile(
+		`^P` + // P at the start
```

Comment:
The current regex allows a duration string to have a time component (H, M, S) without the preceding T separator (e.g., P1H or P1Y1H). The ISO 8601 standard requires the T to separate the date and time parts.